subcommands/subcommands.go

```diff
@@ -0,0 +1,97 @@
+package subcommands
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/joyent/containerpilot/client"
+	"github.com/joyent/containerpilot/config"
+)
+
+// Subcommand provides a simple object for storing a configured HTTPClient.
+type Subcommand struct {
+	client *client.HTTPClient
+}
+
+// Init initializes the configuration of a Subcommand function and the
+// HTTPClient which they utilize for control plane interaction.
+func Init(configFlag string) (*Subcommand, error) {
+	var socketPath = "/var/run/containerpilot.socket"
+
+	if configFlag != "" {
```

Comment:
In an effort to be robust we're repeating logic -- the `config.LoadConfig` already handles all the cases we're handling here, so if we wrap that we'll be hiding the errors it returns and that'll make for an inconsistent UI. We can just return the error from `LoadConfig`.